quarkus.jdt/com.redhat.quarkus.jdt.core/src/main/java/com/redhat/quarkus/jdt/core/JDTQuarkusManagerForJava.java

```diff
@@ -0,0 +1,382 @@
+				if (!url.toString().isEmpty() && path.charAt(0) != '/' && url.charAt(url.length() - 1) != '/') {
+					url.append('/');
+				}
+				url.append(path);
+			}
+		}
+		return url.toString();
+	}
+
+	private static String getJaxRsPathValue(IAnnotatable annotatable) throws JavaModelException {
+		IAnnotation annotationPath = getAnnotation(annotatable, JAVAX_WS_RS_PATH);
+		return annotationPath != null ? getAnnotationMemberValue(annotationPath, PATH_VALUE) : null;
+	}
+
+	private boolean overlaps(ISourceRange typeRange, ISourceRange methodRange) {
+		if (typeRange == null || methodRange == null) {
+			return false;
+		}
+		// method range is overlapping if it appears before or actually overlaps the
+		// type's range
+		return methodRange.getOffset() < typeRange.getOffset() || methodRange.getOffset() >= typeRange.getOffset()
+				&& methodRange.getOffset() <= (typeRange.getOffset() + typeRange.getLength());
+	}
+
+	private static CodeLens createCodeLens(IJavaElement element, ITypeRoot typeRoot, IJDTUtils utils)
+			throws JavaModelException {
+		ISourceRange r = ((ISourceReference) element).getNameRange();
+		if (r == null) {
+			return null;
+		}
+		CodeLens lens = new CodeLens();
+		final Range range = utils.toRange(typeRoot, r.getOffset(), r.getLength());
+		lens.setRange(range);
+		String uri = utils.toClientUri(utils.toUri(typeRoot));
+		lens.setData(Arrays.asList(uri, range.getStart()));
+		return lens;
+	}
+
+	private static boolean isServerAvailable(String host, int port, int timeout) {
+		try (Socket socket = new Socket()) {
+			socket.connect(new InetSocketAddress(host, port), timeout);
+			return true;
+		} catch (IOException e) {
+			return false;
+		}
+	}
+	
+	/**
+	 * Returns the hover information according to the given <code>params</code>
+	 * 
+	 * @param params  the hover parameters
+	 * @param utils   the utilities class
+	 * @param monitor the monitor
+	 * @return the hover information according to the given <code>params</code>
+	 * @throws JavaModelException
+	 */
+	public QuarkusJavaHoverInfo hover(QuarkusJavaHoverParams params, IJDTUtils utils,
+			IProgressMonitor monitor) throws JavaModelException {
+		utils.waitForLifecycleJobs(monitor);
+
+		String uri = params.getUri();
+		final ICompilationUnit unit = utils.resolveCompilationUnit(uri);
+		IClassFile classFile = null;
+		if (unit == null) {
+			classFile = utils.resolveClassFile(uri);
+			if (classFile == null) {
+				return null;
+			}
+		} else {
+			if (!unit.getResource().exists() || monitor.isCanceled()) {
+				return null;
+			}
+		}
+		ITypeRoot typeRoot = unit != null ? unit : classFile;
+		int hoverPositionInt = PositionUtils.positionToInteger(typeRoot, params.getPosition());
+
+		IJavaElement hoverElement = typeRoot.getElementAt(hoverPositionInt);
+
+		if (hoverElement.getElementType() == IJavaElement.FIELD) {
+
+			IField hoverField = (IField) hoverElement;
+
+			IAnnotation annotation = getAnnotation(hoverField, "ConfigProperty");

```

Comment:
use QuarkusConstants.CONFIG_PROPERTY_ANNOTATION insteand of using "ConfigProperty"